optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/OptaPlannerProcessor.java

```diff
@@ -432,6 +432,8 @@ protected void applyScoreDirectorFactoryProperties(IndexView indexView, SolverCo
 
         if (solverConfig.getScoreDirectorFactoryConfig().getScoreDrlList() != null) {
             boolean isKogitoExtensionPresent = capabilities.isPresent("kogito-rules");
+            // Rules do not fire when Drools Alpha Network Compilation is enabled
+            solverConfig.getScoreDirectorFactoryConfig().setDroolsAlphaNetworkCompilationEnabled(false);
```

Comment:
I am not sure why this should happen with DRL and not with CS-D. Both use ANC.
ANC has to be disabled in native mode, because there it doesn't work. But I would really like to understand why it works for non-native CS-D and apparently not for non-native DRL.

Reply:
I would like to know to. A rule like
```
rule "always fire"
    eval(true)
then
    ...
end
```
will always fire whereas
```
rule "never fires"
    Object()
then
    ...
end
```
never fires (despite the fact it should fire for every fact). It does this silently; no errors -- it can only be caught using an assertion score director or looking at the solution.

Reply:
Weird indeed. Is ANC on the classpath? Could be useful to hook up debugger to Drools and check what happens when facts are inserted into the working memory. Maybe nothing happens because nothing in inserted?

Although I think that it is more likely that there is some classloader magic somewhere and this `Object` isn't the same as the `Object` being inserted.

Reply:
Could the ANC be happening with a different classloader than what Quarkus uses? (Just guessing.)

Reply:
It might; it uses the root class loader of the InternalKnowledgeBase.

Reply:
`KieBaseUpdaterANC.generateAndSetInMemoryANC(kieBase)` works with `kogito-quarkus-rules` with Drools by itself.